packages/pricing/src/models/money-amount.ts

```diff
@@ -1,67 +1,67 @@
-import { DALUtils, generateEntityId } from "@medusajs/utils"
+import { DAL } from "@medusajs/types"
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 import {
   BeforeCreate,
-  Collection,
   Entity,
   Filter,
-  Index,
-  ManyToMany,
   OnInit,
   OneToOne,
   OptionalProps,
   PrimaryKey,
   Property,
 } from "@mikro-orm/core"
-
 import { PriceSetMoneyAmount } from "./index"
-import PriceSet from "./price-set"
 
-@Entity()
+type OptionalFields = DAL.SoftDeletableEntityDateColumns
+
+const tableName = "money_amount"
+const MoneyAmountDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+
+const MoneyAmountCurrencyCodeIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "currency_code",
+  where: "deleted_at IS NULL",
+})
+
+@Entity({ tableName })
 @Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
 class MoneyAmount {
-  [OptionalProps]?:
-    | "created_at"
-    | "updated_at"
-    | "deleted_at"
-    | "price_set_money_amount"
-    | "amount"
+  [OptionalProps]?: OptionalFields
 
   @PrimaryKey({ columnType: "text" })
   id!: string
 
-  @Property({
-    columnType: "text",
-    nullable: true,
-    index: "IDX_money_amount_currency_code",
-  })
-  currency_code: string | null
-
-  @ManyToMany({
-    entity: () => PriceSet,
-    mappedBy: (ps) => ps.money_amounts,
-  })
-  price_sets = new Collection<PriceSet>(this)
-
-  @OneToOne({
-    entity: () => PriceSetMoneyAmount,
-    mappedBy: (psma) => psma.money_amount,
-    cascade: ["soft-remove"] as any,
-  })
-  price_set_money_amount: PriceSetMoneyAmount
+  @MoneyAmountCurrencyCodeIndex.MikroORMIndex()
+  @Property({ columnType: "text" })
+  currency_code: string
 
   @Property({
     columnType: "numeric",
-    nullable: true,
     serializer: Number,
   })
-  amount: number | null
+  amount: number
```

Comment:
Q: should we use big numbers?

Reply:
Good idea. I'll open a PR for this specifically.